src/sp/it/pl/gui/objects/contextmenu/ImprovedContextMenu.kt

```diff
@@ -118,17 +119,20 @@ class ContextMenuBuilder<T>(val contextMenu: ImprovedContextMenu<*>, val selecte
 
     private val items = ArrayList<MenuItem>()
 
-    fun getItemSequence() = items.asSequence()
+    fun asSequence() = items.asSequence()
 
-    private fun <T: MenuItem> T.add() = also { items.add(it) }
+    fun <T: MenuItem> item(item: T): T {
```

Comment:
I still have last remaining question:

Why not provide menu() method for menus? I know Menu is subclass of MenuItem, but for the builder to be declarative, wouldnt it be better to disambiguate Menu and MenuItem at all times?

Btw: the fact that these methods return the constructed item is slightly bothering me as well. There should be an optional initialization lambda parameter and menu()/item() should return Unit. The only problem with this is that if the item(item: MenuItem) is not infix we get:
```this.item(myMenuItem.apply { ... })```
instead of:
```this.item myMenuItem.apply { ... }```
Unfortunately, making the menu infix either requires this. at callsite or special magic which will make item() not an extension function but extension property, at which point the dsl becomes unstructured again. A possible fix would be to turn all menu/item extension functions into extension properties. It is possible. Please see the implementation of ```Pane.lay``` if you are interested in how.

This is all over-thinking it, its good as it is.

Reply:
Okay, so are you gonna approve it now?

One idea of that method was that there could be other subclasses of MenuItem, it should be open. Also, the actual method is never used outside of the extension methods.